extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/quantiles/DoublesSketchToHistogramPostAggregator.java

```diff
@@ -125,14 +155,26 @@ public boolean equals(final Object o)
     if (!Arrays.equals(splitPoints, that.splitPoints)) {
       return false;
     }
-    return field.equals(that.field);
+    if (!field.equals(that.field)) {
+      return false;
+    }
+    if (numBins == null && that.numBins == null) {
```

Comment:
Please add an EqualsVerifier test too. We don't have them everywhere yet but we're trying to add them in patches that adjust equals or hashCode implementations. Check out InFilterTest for an example. It looks like this.

```
  @Test
  public void testEquals()
  {
    EqualsVerifier.forClass(InFilter.class)
                  .usingGetClass()
                  .withNonnullFields("dimension", "values")
                  .withIgnoredFields("longPredicateSupplier", "floatPredicateSupplier", "doublePredicateSupplier")
                  .verify();
  }
```

Reply:
It seems to me that this EqualsVerifier is not visible in this branch. Perhaps it was added to some pom file later. Would you like me to merge from master or rebase?

Reply:
You might need to add this to the datasketches pom:

```xml
    <dependency>
      <groupId>nl.jqno.equalsverifier</groupId>
      <artifactId>equalsverifier</artifactId>
      <scope>test</scope>
    </dependency>
```

Reply:
Merging from master would be useful too if your branch is old enough that it doesn't have `equalsverifier` in the master pom.